### PR TITLE
fix: apply multiline inline table patching fixes

### PR DESCRIPTION
**Pull request by DecimalTurn**
This pull request adds and improves support for preserving formatting and comments in TOML v1.1 multiline inline tables, especially when editing or removing keys and objects. It introduces new tests to cover edge cases and refines the logic for handling inline tables and comments during patching and removal operations.

Main fixes:

1) Nested tables with the following structure: inline table > inline array > inline table can be patched:

```toml
      items = [
        {
          name = "Hammer",
          color = "red",
        },
        {
          name = "Nail",
          color = "gray",
        }
      ]
```

2) Full comments preservation when deleting elements.

In regular key-value pairs in regular tables, we expect that if we remove the kvp, it should remove the comment on the same line. However, for inline tables, it's not clear what the expectations should be. For instance, consider the following:

```toml
	demo = {
		a = "test1", b = "test2" # This is a comment
	}
```

Should deleting `b = "test2"` means that we should also remove the comment at the end of the line? What if the comment was for both `a` and `b` and remains valid information after `b` is removed? Due to this ambiguity, we are better off not deleting comments inside inline tables. This PR should ensure that's the case.

### CODE REVIEW

File: src/writer.ts [hunk 1]
```diff
@@ -490,7 +490,17 @@ export function remove(root: Root, parent: TreeNode, node: TreeNode) {
...
+    (root as WithItems).items.some(item => isComment(item));
+
+  if (
+    previous === undefined &&
+    next === undefined &&
+    !hasRootInlineTableComments
+  ) {
     offset.lines = 0;
     offset.columns = 0;
   }
```
[the remark this answers is not in the document]
Comment (DecimalTurn) [reply src/writer.ts]: b4805c9b01a3fdca9d416f686a34008afa430cd8

File: src/writer.ts [hunk 2]
```diff
@@ -490,7 +490,17 @@ export function remove(root: Root, parent: TreeNode, node: TreeNode) {
...
+    isInlineTable(parent) &&
+    hasItems(root) &&
+    root !== parent &&
+    (root as WithItems).items.some(item => isComment(item));
+
+  if (
+    previous === undefined &&
+    next === undefined &&
+    !hasRootInlineTableComments
+  ) {
```
[the remark this answers is not in the document]
Comment (DecimalTurn) [reply src/writer.ts]: b4805c9b01a3fdca9d416f686a34008afa430cd8

[the remark this answers is not in the document]
Comment (DecimalTurn) [reply src/writer.ts]: 7cc9905f36aa4343a1a628d6ec1490fa3d3aeea8

File: src/writer.ts [hunk 3]
```diff
@@ -582,7 +592,7 @@ export function remove(root: Root, parent: TreeNode, node: TreeNode) {
       if (commentLine === removedLine) {
         // Comment was on the same line as the removed item — drop it.
         toRemove.push(i);
-      } else if (commentLine < removedLine) {
+      } else if (offset.lines !== 0 && commentLine < removedLine) {
         // Comment is before the removed line: pre-compensate so the bleedthrough
```
[the remark this answers is not in the document]
Comment (DecimalTurn) [reply src/writer.ts]: 7cc9905f36aa4343a1a628d6ec1490fa3d3aeea8
